src/OpenTelemetry.Exporter.Prometheus/Implementation/PrometheusExporterMiddleware.cs

```diff
@@ -79,7 +79,9 @@ public async Task InvokeAsync(HttpContext httpContext)
                     }
                     else
                     {
-                        throw new InvalidOperationException("Collection failure.");
+                        // It's not expected to have no metrics to collect, but it's not necessarily a failure, either.
+                        response.StatusCode = 204;
```

Comment:
I have a slight concern here that clients might not know what to do with this. I checked a few spots, didn't see anywhere where status code was being defined, other than 200.

* prometheus-net: https://github.com/prometheus-net/prometheus-net/blob/ff337560e66af5dd45070014d6e554febb599f16/Prometheus.AspNetCore/MetricServerMiddleware.cs#L38
* OTel spec: https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk_exporters/prometheus.md
* Prometheus docs: https://github.com/prometheus/docs/blob/main/content/docs/instrumenting/exposition_formats.md

I'm going to approve because it is probably rare people will have no metrics beyond their initial setup, but it would make me feel better if it was defined somewhere 😄

Reply:
+1, I think this is something https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk_exporters/prometheus.md should define (currently it's Experimental).

Reply:
204 seemed _reasonable_ since it was literally zero content coming back. If it needs to change, someone let me know what content the 200 would return, since 200 does need a content type and... like... some content. 😄 

Note the goal here was to allow a Kubernetes pod to start up and have Prometheus _not_ report the scrape endpoint as an error in connectivity, so having some sort of success status code is the real key here. Whether it's 200 with some sort of header-only content or 204 with no content makes no difference to me. Let me know.